docs/en/routing/basic-usage.md

```diff
@@ -0,0 +1,490 @@
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={() => <div>Home</div>} />
+			</div>
+		);
+	}
+}
+```
+
+## Path & Query Parameters
+
+Path parameters are placeholders in the routing configuration that will match any value for the segment. The parameters are defined using curly braces, for example: `{param}`.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home'
+	}
+];
+```
+
+The parameters values are injected into to matching `Outlets`'s `renderer` property.
+
+>src/App.tsx
+```tsx
+import WidgetBase from '@dojo/framework/widget-core/WidgetBase';
+import { tsx } from '@dojo/framework/widget-core/tsx';
+import Outlet from '@dojo/framework/routing/Outlet';
+
+export default class App extends WidgetBase {
+	protected render() {
+		return (
+			<div>
+				<Outlet id="home" renderer={(matchDetails) => <div>{`Home ${matchDetails.params.page}`}</div>} />
+			</div>
+		);
+	}
+}
+```
+
+Query parameters are also defined using curly braces split from the route path by a `?`. To define more than one query param, use the `&` delimiter.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}?{queryOne}&{queryOne}',
+		outlet: 'home'
+	}
+];
+```
+
+The query params are injected into matching `Outlet` `renderer` properties accessed from `queryParams` property.
+
+## Default Route & Parameters
+
+To add a default route for your application update the routing configuration to include `defaultRoute: true` for the preferred route.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home',
+		outlet: 'home',
+		defaultRoute: true
+	}
+];
+```
+
+If the default route has path or query parameters a map of defaults need to be specified.
+
+>src/routes.ts
+```ts
+export default [
+	{
+		path: 'home/{page}',
+		outlet: 'home',
+		defaultRoute: true,
+		defaultParams: {
+			page: 'about'
+		}
+	}
+];
+```
+
+## Route Configuration
+
+The routing configuration is a hierarchical structure used to describe the entire Dojo application, associating `outlet` ids to a routing path. The routing path can be nested using children which enables building a routing structure that can accurately reflect the requirements of the application.
+
+The routing configuration API is constructed with the following properties:
+
+* `path: string`: The routing path segment to match in the URL.
+* `outlet: string`: The `outlet` id used to render widgets to the associated routing path.
+* `defaultRoute: boolean`(optional): Marks the outlet as default, the application will redirect to this route automatically if an unknown routes is detected on application load.
```

Comment:
So here's the description of what a default route is - might be worth calling this out & linking to it from the basic usage sample?

Reply:
> Marks the outlet as default, the application will redirect to this route automatically if an unknown routes is detected on application load.

Doesn't it also redirect if no route provided?

Reply:
Correct, I'll add that @matt-gadd specifically - technically no route is an unknown route but it's not obvious.

Reply:
Updated